src/main/java/gregtech/common/crafting/ToolHeadReplaceRecipe.java

```diff
@@ -33,30 +33,30 @@ public boolean matches(@Nonnull InventoryCrafting inv, @Nonnull World world) {
         List<ItemStack> list = new ArrayList<>();
 
         for (int i = 0; i < inv.getSizeInventory(); i++) {
-            ItemStack itemstack = inv.getStackInSlot(i);
+            ItemStack stack = inv.getStackInSlot(i);
+            if (!stack.isEmpty()) {
+                list.add(stack);
+            }
```

Comment:
You could introduce an early-exit case here, where if the list contained more than two items, immediately return false.